Ginger/Ginger/ALM/Qtest/TreeViewItems/QtestCycleTreeItem.cs

```diff
@@ -0,0 +1,104 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+See the License for the specific language governing permissions and 
+limitations under the License. 
+*/
+#endregion
+
+using Ginger.SolutionWindows.TreeViewItems;
+using GingerWPF.UserControlsLib.UCTreeView;
+using GingerCore;
+using System;
+using System.Collections.Generic;
+using System.Windows.Controls;
+using Amdocs.Ginger.Common;
+using amdocs.ginger.GingerCoreNET;
+using GingerCore.ALM.Qtest;
+
+namespace Ginger.ALM.Qtest.TreeViewItems
+{
+    public class QtestCycleTreeItem : TreeViewItemBase, ITreeViewItem
+    {
+        public BusinessFlow QCExplorer { get; set; }
+        public List<ITreeViewItem> currentChildrens = null;
+
+        public string ID { get; set; }
+        public string Name { get; set; }
+        public string Path { get; set; }
+
+        Object ITreeViewItem.NodeObject()
+        {
+            return null;
+        }
+
+        public QtestCycleTreeItem(List<QTestApiModel.TestCycleResource> childrenCycles = null, List<QTestApiModel.TestSuiteWithCustomFieldResource> childrenSuites = null)
+        {
+            currentChildrens = new List<ITreeViewItem>();
+
+            if (childrenCycles != null)
+            {
+                foreach (QTestApiModel.TestCycleResource cycle in childrenCycles)
+                {
+                    QtestCycleTreeItem pfn = new QtestCycleTreeItem(cycle.TestCycles, cycle.TestSuites);
+                    pfn.Name = cycle.Name.ToString();
+                    pfn.ID = cycle.Id.ToString();
+                    pfn.Path = cycle.Path.ToString();
+                    currentChildrens.Add(pfn);
+                }
+            }
+
+            if (childrenSuites != null)
+            {
+                foreach (QTestApiModel.TestSuiteWithCustomFieldResource suite in childrenSuites)
+                {
+                    QtestSuiteTreeItem pfn = new QtestSuiteTreeItem();
+                    pfn.Name = suite.Name.ToString();
+                    pfn.ID = suite.Id.ToString();
+                    currentChildrens.Add(pfn);
+                }
+            }
+        }
+
+        StackPanel ITreeViewItem.Header()
+        {
+            return TreeViewUtils.CreateItemHeader(Name, "@WorkFlow_16x16.png");
+            // return TreeViewUtils.CreateItemHeader(Name, "@TestSet_16x16.png");
+            // return TreeViewUtils.CreateItemHeader(Name, "@Folder_16x16.png");
+        }
+
+        List<ITreeViewItem> ITreeViewItem.Childrens()
+        {
+            return currentChildrens;
+        }
+
+        bool ITreeViewItem.IsExpandable()
+        {
+            return true;
+        }
+
+        Page ITreeViewItem.EditPage(Amdocs.Ginger.Common.Context mContext)
+        {
+            return null;
+        }
+
+        ContextMenu ITreeViewItem.Menu()
+        {
+            return null;
+        }
+        
+        void ITreeViewItem.SetTools(ITreeView TV)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a nested comment explaining why this method is empty, throw a 'NotSupportedException' or complete the implementation.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)